airflow/upgrade/checker.py

```diff
@@ -41,6 +41,15 @@ def check_upgrade(formatter, rules):
     return all_rule_statuses
 
 
+def list_checks():
+    print()
+    print("Upgrade Checks:")
+    for rule in ALL_RULES:
```

Comment:
```suggestion
    for rule in sorted(ALL_RULES, key=lambda r: r.__class__.__name__:
```

This will introduce a nice order. In fact we can even use this `sorted` in definition of `ALL_RULES` to keep it sorted everywhere, WDYT?

Reply:
It looks like the current assumption is that the version check should run first (based off of the fact that these files are run in alphabetical order and it's currently prefixed with `aaa_`: https://github.com/apache/airflow/tree/v1-10-stable/airflow/upgrade/rules). This will alter that order, and I think it might not be best to change the name of the version check class as people may already be ignoring it/using it elsewhere. I'm open to changing this if those concerns aren't as widely shared :)

Reply:
@madison-ookla how about introducing order here:
https://github.com/apache/airflow/blob/748d05f519c0e5427d2a46763591f03697c4d585/airflow/upgrade/rules/__init__.py#L20-L34

by doing:
```py
return rule_classes[0] + sorted(rule_classes[1:],  key=lambda r: r.__class__.__name__)
```
we can assume that check version rule will be first (the files are sorted so `aaa_` will be first) and then we can easily sort it.

Reply:
Sounds good!

Reply:
Just pushed this change, new output (and ordering) looks like this:

```

Upgrade Checks:
- Check for latest versions of apache-airflow and checker [VersionCheckRule]
- Remove airflow.AirflowMacroPlugin class [AirflowMacroPluginRemovedRule]
- Ensure users are not using custom metaclasses in custom operators [BaseOperatorMetaclassRule]
- Chain between DAG and operator not allowed. [ChainBetweenDAGAndOperatorNotAllowedRule]
- Connection.conn_type is not nullable [ConnTypeIsNotNullableRule]
- Hooks that run DB functions must inherit from DBApiHook [DbApiRule]
- Fernet is enabled by default [FernetEnabledRule]
- GCP service account key deprecation [GCPServiceAccountKeyRule]
- Unify hostname_callable option in core section [HostnameCallable]
- Changes in import paths of hooks, operators, sensors and others [ImportChangesRule]
- Legacy UI is deprecated by default [LegacyUIDeprecated]
- Logging configuration has been moved to new section [LoggingConfigurationRule]
- Removal of Mesos Executor [MesosExecutorRemovedRule]
- No additional argument allowed in BaseOperator. [NoAdditionalArgsInOperatorsRule]
- Users must set a kubernetes.pod_template_file value [PodTemplateFileRule]
- SendGrid email uses old airflow.contrib module [SendGridEmailerMovedRule]
- Changes in import path of remote task handlers [TaskHandlersMovedRule]
- Jinja Template Variables cannot be undefined [UndefinedJinjaVariablesRule]
- Connection.conn_id is not unique [UniqueConnIdRule]
```